### PR TITLE
[Sketcher] Slot DSH : implement tool parameters.

### PR DESCRIPTION
**Pull request by PaddleStroke**
Implement tool parameters for the Slot tool in sketcher.
@abdullahtahiriyo

### CODE REVIEW

**Comment by PaddleStroke** [comment]
Lint warning to fix : virtual to remove. I'll do it later.

**Comment by PaddleStroke** [comment]
This is ready too. Arcslot too.

File: src/Mod/Sketcher/Gui/DrawSketchHandlerSlot.h [hunk 1]
```diff
@@ -44,333 +46,540 @@ namespace SketcherGui
...
-                    start = 1.5 * M_PI;
-                    end = 0.5 * M_PI;
-                }
+        ShapeGeometry.clear();
+
+        if (length < Precision::Confusion() || radius < Precision::Confusion()) {
+            return;
+        }
+
+        auto arc1 = std::make_unique<Part::GeomArcOfCircle>();
```
Comment (abdullahtahiriyo) [inline src/Mod/Sketcher/Gui/DrawSketchHandlerSlot.h:257]: This is the kind of code that should use the convenience functions ```addArcToShapeGeometry```a similar.

File: src/Mod/Sketcher/Gui/DrawSketchHandlerSlot.h [hunk 2]
```diff
@@ -44,333 +46,540 @@ namespace SketcherGui
...
+            onViewParameters[OnViewParameter::Second]->setPoints(Base::Vector3d(),
+                                                                 toVector3d(onSketchPos));
+        } break;
+        case SelectMode::SeekSecond: {
+            Base::Vector3d start = toVector3d(handler->startPoint);
+            Base::Vector3d end = toVector3d(handler->secondPoint);
+            Base::Vector3d vec = end - start;
+
+            if (!onViewParameters[OnViewParameter::Third]->isSet) {
+                onViewParameters[OnViewParameter::Third]->setSpinboxValue(vec.Length());
```
Comment (abdullahtahiriyo) [inline src/Mod/Sketcher/Gui/DrawSketchHandlerSlot.h:445]: Framework.

**Comment by abdullahtahiriyo** [comment]
And cursors.

**Comment by abdullahtahiriyo** [comment]
@PaddleStroke 

This is not ready, is it?

**Comment by PaddleStroke** [comment]
I'm wraping up, in 2 minutes yes

**Comment by PaddleStroke** [comment]
@abdullahtahiriyo That's it I just pushed.

**Comment by abdullahtahiriyo** [comment]
We have got an issue here:

[Peek 2023-11-13 15-38.webm](https://github.com/FreeCAD/FreeCAD/assets/5249162/7021cd3f-41e5-47a2-a14a-1b6e250b80fc)

**Comment by PaddleStroke** [comment]
So there was actually multiple bugs in one. Everyone tries to set horizontal/vertical. I think I nailed it down.

**Pull request merged**